Migrate labeler config to new format

adegeo opened the pull request:
Converts the old labeler tool config file into the new tool rule file.

guardrex commented:
The ones for ...

* `"(?i).*main\/aspnetcore\/security\/blazor.*"`
* `"(?i).*main\/aspnetcore\/host-and-deploy\/blazor.*"`

... can be 🔪 cut ... those nodes moved into the `aspnetcore/blazor` folder, so they get their labels from the earlier entry.

guardrex commented:
btw ... Can this do other things to a new issue?

Can it ...

* Add an assignee?
* Subtract a label?
* Add the issue to a GH project?

adegeo commented:
@guardrex 
It has the potential to do any of that yes. I recently added projects support (via id). It doesn't mess with assignee's yet. I just rewrote the tool from scratch to clean up the code (since it was 3 revisions deep of globbing functionality on top of itself and redesigning things) Now it's very easy for me to add new command support. Currently it can trigger off of any event associated with issues, prs, comments, and can react with the following actions:

- Label (add,remove)
- Project (add)
- Milestone (set, remove)

It supports conditionals via
- JMESPath query on the data for the issue/pr/comment/payload
- The MS.DOCS field information (content path, author) in the original comment
- Pull request file changed regex

You can see a bigger implementation on the [dotnet docs rules pr](https://github.com/dotnet/docs/pull/24618/files#diff-89138f5a7fbe726c84b8bafd0dbf0f4582236675a7c4226187fbfd92a03dbaef)

guardrex commented:
Thanks @adegeo ... Should I just circle around after you get this merged, make the proposed updates (for the ones it supports, i.e., not with `assignees`), and then ping u on the PR?

guardrex commented:
Ok, great. This will be *very* helpful, as I'm having to add these to **_hundreds_** of Blazor issues.

adegeo commented:
If the docs have the appropriate metadata fields, would it be easier to remove the regex check on file path and instead use a single check of `technology = aspnetcore-blazor` ?


![image](https://user-images.githubusercontent.com/67293991/121555422-dab67e80-c9c7-11eb-8ce8-2b7728ee0691.png)

adegeo commented:
Same for 

![image](https://user-images.githubusercontent.com/67293991/121555679-0fc2d100-c9c8-11eb-9000-51969e04049d.png)

guardrex commented:
Yes, indeed. That's fine with me if @Rick-Anderson is ok with it. They do some internal tracking I believe, so he'll have to say if that would negatively affect anything.

guardrex commented:
WRT adding the tech metadata to the Blazor topics, I'll take care of that if it's approved.

adegeo commented:
It might be already be in the articles, since it shows up in the issue.

guardrex commented:
Oh ... yes ... you are correct. Scott took care of that ages ago. I forgot. :smile:

Rick-Anderson commented:
good plan

adegeo commented:
@Rick-Anderson @guardrex I updated the rules.

- Added 3 rules that trigger off of an ms.docs issue with a specific technology category, adds a label
- Kept the specific file path to a single file that adds a label
- Added 3 new rules for when the word (blazor,signalr,grpc) appears in the file name to add a label

guardrex commented:
Can you have it take off the `Not Triaged` label for the Blazor issues? That's one that I have to manually delete. The default column for a Blazor issue is the `Triaged` column, so it's a redundant label in terms of my workflow.

adegeo commented:
@guardrex this has been merged. Let me know if something isn't working. Cheers!
